File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/menuItems/MenuItem.ts
import { Language } from "../Language";
import { PriceType } from "./PriceType";

export interface MenuItem {
    readonly id: string;
    readonly name: string;
    readonly description?: string;
    readonly price: number;
    readonly priceType: PriceType;
    readonly locationId?: string;
    readonly showWhenNotAvailable: boolean;
    readonly sortIndex: number;
    readonly stock: boolean;
    readonly vatRate: number;
    readonly imageUrl?: string;
    readonly translations: Record<Language, MenuItemTranslation>;
    readonly menuCategoryIds: string[];
}

export interface MenuItemTranslation {
    readonly name: string;
    readonly description?: string;
}

File: Quivi.Backoffice.Api/Dtos/MenuItem.cs
using Quivi.Domain.Entities;
using Quivi.Domain.Entities.Pos;

namespace Quivi.Backoffice.Api.Dtos
{
    public class MenuItem
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public string? Description { get; init; }
        public string? ImageUrl { get; init; }
        public bool Stock { get; init; }
        public decimal Price { get; init; }
        public PriceType PriceType { get; init; }
        public bool ShowWhenNotAvailable { get; init; }
        public decimal VatRate { get; init; }
        public int SortIndex { get; init; }
        public string? LocationId { get; init; }
        public required IEnumerable<string> MenuCategoryIds { get; init; }
        public required IEnumerable<string> ModifierGroupIds { get; init; }
        public required IReadOnlyDictionary<Language, MenuItemTranslation> Translations { get; init; }
    }

    public class MenuItemTranslation
    {
        public required string Name { get; init; }
        public string? Description { get; init; }
    }
}
